fulp_modules/bloodsuckers/bloodsucker/bloodsucker_coffin.dm

```diff
@@ -0,0 +1,176 @@
+/datum/antagonist/bloodsucker/proc/ClaimCoffin(obj/structure/closet/crate/claimed) // NOTE: This can be any "closet" that you are resting AND inside of.
+	// ALREADY CLAIMED
+	if(claimed.resident)
+		if(claimed.resident == owner.current)
+			to_chat(owner, "This is your [src].")
+		else
+			to_chat(owner, "This [src] has already been claimed by another.")
+		return FALSE
+	// Bloodsucker Learns new Recipes!
+	owner.teach_crafting_recipe(/datum/crafting_recipe/vassalrack)
+	owner.teach_crafting_recipe(/datum/crafting_recipe/candelabrum)
+	owner.teach_crafting_recipe(/datum/crafting_recipe/meatcoffin)
+	// This is my Lair
+	coffin = claimed
+	lair = get_area(claimed)
+	// DONE
+	to_chat(owner, "<span class='userdanger'>You have claimed the [claimed] as your place of immortal rest! Your lair is now [lair].</span>")
+	to_chat(owner, "<span class='danger'>You have learned new construction recipes to improve your lair.</span>")
+	to_chat(owner, "<span class='announce'>Bloodsucker Tip: Find new lair recipes in the Misc tab of the <i>Crafting Menu</i>, including the <i>Persuasion Rack</i> for converting crew into Vassals.</span><br><br>")
+	RunLair() // Start
+	return TRUE
+
+// crate.dm
+/obj/structure/closet/crate
+	var/mob/living/resident	// This lets bloodsuckers claim any "closet" as a Coffin, so long as they could get into it and close it. This locks it in place, too.
+	var/pryLidTimer = 250
+	can_weld_shut = FALSE
+	breakout_time = 200
+
+/obj/structure/closet/crate/coffin/blackcoffin
+	name = "black coffin"
+	desc = "For those departed who are not so dear."
+	icon_state = "coffin"
+	icon = 'fulp_modules/bloodsuckers/icons/vamp_obj.dmi'
+	open_sound = 'fulp_modules/bloodsuckers/sounds/coffin_open.ogg'
+	close_sound = 'fulp_modules/bloodsuckers/sounds/coffin_close.ogg'
+	breakout_time = 600
+	pryLidTimer = 400
+	resistance_flags = NONE
+	material_drop = /obj/item/stack/sheet/metal
+	material_drop_amount = 2
+	armor = list("melee" = 50, "bullet" = 20, "laser" = 30, "energy" = 0, "bomb" = 50, "bio" = 0, "rad" = 0, "fire" = 70, "acid" = 60)
+
+/obj/structure/closet/crate/coffin/meatcoffin
+	name = "meat coffin"
+	desc = "When you're ready to meat your maker, the steaks can never be too high."
+	icon_state = "meatcoffin"
+	icon = 'fulp_modules/bloodsuckers/icons/vamp_obj.dmi'
+	resistance_flags = FIRE_PROOF
+	open_sound = 'sound/effects/footstep/slime1.ogg'
+	close_sound = 'sound/effects/footstep/slime1.ogg'
+	breakout_time = 200
+	pryLidTimer = 200
+	resistance_flags = NONE
+	material_drop = /obj/item/food/meat/slab/human
+	material_drop_amount = 3
+	armor = list("melee" = 70, "bullet" = 10, "laser" = 10, "energy" = 0, "bomb" = 70, "bio" = 0, "rad" = 0, "fire" = 70, "acid" = 100)
```

Comment:
dunno why this is immune to acid; ain't acid pretty good at burning through flesh?

an acid-immune coffin does sound pretty nice, though